src/m365/pp/commands/dataverse/dataverse-webresource-list.ts

```diff
@@ -0,0 +1,201 @@
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environmentName <environmentName>'
+      },
+      {
+        option: '-s, --solutionId <solutionId>'
+      },
+      {
+        option: '--excludeContent'
+      },
+    );
+  }
+
+  private WEBRESOURCE_PROPS_ALL = [
+    'webresourceid',
+    'name',
+    'canbedeleted',
+    'componentstate',
+    'content',
+    'content_binary',
+    'contentfileref',
+    'contentjson',
+    'contentjsonfileref',
+    'createdon',
+    'dependencyxml',
+    'description',
+    'displayname',
+    'introducedversion',
+    'isavailableformobileoffline',
+    'iscustomizable',
+    'isenabledformobileclient',
+    'ishidden',
+    'ismanaged',
+    'languagecode',
+    'modifiedon',
+    'overwritetime',
+    'silverlightversion',
+    'solutionid',
+    'versionnumber',
+    'webresourceidunique',
+    'webresourcetype'
+  ];
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    await this.logIfVerbose(logger, `Retrieving list of web resources for solution '${args.options.solutionId}' in environment '${args.options.environmentName}'...`);
+
+    try {
+      const dynamicsApiUrl = await powerPlatform.getDynamicsInstanceApiUrl(args.options.environmentName);
+
+      this.logIfVerbose(logger, `Found Dynamics API URL: ${dynamicsApiUrl}`);
+
+      if (await this.checkIfSolutionExists(args.options.solutionId, dynamicsApiUrl, logger) === false) {
+        throw new Error(`Solution with ID '${args.options.solutionId}' not found in environment '${args.options.environmentName}'`);
+      }
+
+      const solutionComponentsRequestUrl = `${dynamicsApiUrl}/api/data/v9.2/solutioncomponents?$filter=_solutionid_value eq ${args.options.solutionId} and componenttype eq 61&$select=objectid`;
+
+      this.logIfVerbose(logger, `Retrieving web resources solution components`);
+      const solutionComponentResponse = await odata.getAllItems<SolutionComponent>(solutionComponentsRequestUrl);
+
+      this.logIfVerbose(logger, `Found ${solutionComponentResponse.length} web resource solution component in solution '${args.options.solutionId}' in environment '${args.options.environmentName}'`);
+      const props = this.getWebresourceProperties(args);
+
+      this.logIfVerbose(logger, `Using props: ${props.join(',')}`);
+      const webResourcesRequestUrl = `${dynamicsApiUrl}/api/data/v9.2/webresourceset?$select=${props.join(',')}&$filter=${solutionComponentResponse.map((c: SolutionComponent) => `webresourceid eq ${c.objectid}`).join(' or ')}`;
+
+      this.logIfVerbose(logger, `Retrieving web resources from URL: ${webResourcesRequestUrl}`);
+      const webResourcesResponse = await odata.getAllItems<Webresource>(webResourcesRequestUrl);
+
+      if (!args.options.output || !cli.shouldTrimOutput(args.options.output)) {
```

Comment:
I don't think we have to do an output check. We can just return the result below for all output types. CLI for Microsoft 365 will automatically take the `defaultProperties` into account if they are in the root of the object. We only have to check for output types if the specified default properties are not in the root of the object and we have to do some manipulations.